src/fides/api/ops/models/privacy_notice.py

```diff
@@ -0,0 +1,191 @@
+    disabled = Column(Boolean, nullable=False, default=False)
+    has_gpc_flag = Column(Boolean, nullable=False, default=False)
+    displayed_in_privacy_center = Column(Boolean, nullable=False, default=True)
+    displayed_in_banner = Column(Boolean, nullable=False, default=True)
+    displayed_in_privacy_modal = Column(Boolean, nullable=False, default=True)
+
+
+class PrivacyNotice(PrivacyNoticeBase, Base):
+    """
+    A notice set up by a system administrator that an end user (i.e., data subject)
+    accepts or rejects to indicate their consent for particular data uses
+    """
+
+    def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyNotice:
+        """
+        Overrides the base update method to automatically bump the version of the
+        PrivacyNotice record and also create a new PrivacyNoticeHistory entry
+        """
+        # history record data is identical to privacy notice record data
+        # except the notice's 'id' must be moved to the FK column
+        # and is no longer the history record 'id' column
+        history_data = {
+            "name": self.name,
+            "description": self.description or None,
+            "origin": self.origin or None,
+            "regions": self.regions,
+            "consent_mechanism": self.consent_mechanism,
+            "data_uses": self.data_uses,
+            "enforcement_level": self.enforcement_level,
+            "version": self.version,
+            "disabled": self.disabled,
+            "has_gpc_flag": self.has_gpc_flag,
+            "displayed_in_privacy_center": self.displayed_in_privacy_center,
+            "displayed_in_banner": self.displayed_in_banner,
+            "displayed_in_privacy_modal": self.displayed_in_privacy_modal,
+            "privacy_notice_id": self.id,
+        }
+        PrivacyNoticeHistory.create(db, data=history_data, check_name=False)
+
+        # on any update to a privacy notice record, its version must be incremented
+        # version gets incremented by a full integer, i.e. 1.0 -> 2.0 -> 3.0
+        data["version"] = float(self.version) + 1.0
+        return super().update(db=db, data=data)  # type: ignore
+
+    def dry_update(self, *, data: dict[str, Any]) -> FidesBase:
+        """
+        A utility method to get an updated object without saving it to the db.
+
+        This is used to see what an object update would look like, in memory,
+        without actually persisting the update to the db
+        """
+        # Update our attributes with values in data
+        cloned_attributes = self.__dict__.copy()
+        for key, val in data.items():
+            cloned_attributes[key] = val
+
+        # remove protected fields from the cloned dict
+        cloned_attributes.pop("_sa_instance_state")
+
+        # create a new object with the updated attribute data to keep this
+        # ORM object (i.e., `self`) pristine
+        return PrivacyNotice(**cloned_attributes)
+
+
+def check_conflicting_data_uses(
```

Comment:
hopefully this method is at least easy to follow. there may be some optimizations to make. i'd initially tried to create a 2d/nested dict with both regions and data_uses to allow for indexing on the data_uses for a given region (constant time lookups rather than linear, i.e. iterating thru all data uses for a given region), but ran into issues given that we need to do the "hierarchical" comparison of data uses.

we could probably create some sort of hierarchical data structure that allowed for lookups, but that seemed to be too much complexity for runtime optimizations that i'm assuming are not going to have a significant impact, at this point.

but there may be other improvements here i haven't seen - my main focus was getting something in place that worked and wasn't too painful to follow!